File: 1548-check-if-all-1s-are-at-least-length-k-places-away/check-if-all-1s-are-at-least-length-k-places-away.cpp
class Solution {
public:
    bool kLengthApart(vector<int>& nums, int k) {
        int prev = 0, sz = nums.size();

        while(prev < sz && nums[prev] != 1) prev++;

        int curr = prev+1;
        while(curr < sz){
            while(curr < sz && nums[curr] != 1) curr++;
            if(curr < sz && nums[curr] == 1){
                if(curr - prev - 1 < k) return false;
            }
            prev = curr;
            curr++;
        }

        return true;
    }
};

File: 1548-check-if-all-1s-are-at-least-length-k-places-away/check-if-all-1s-are-at-least-length-k-places-away.java
class Solution {
    public boolean kLengthApart(int[] nums, int k) {
        int sz = nums.length, lastused = -1;

        for(int i=0; i<sz; i++){
            // System.out.println(lastused + " " + i + " " + (i - lastused - 1));
            if(nums[i] == 1 && (lastused == -1 || (i - lastused - 1) >= k)) lastused = i;
            else if(nums[i] == 1) return false;
        }

        return true;
    }
}